### DISCUSSION TITLE
How to select device when using addStyle method

### DISCUSSION THREAD

**Discussion by mauriciolcs2** [GrapesJS/grapesjs#5673] [Q&A] [answered]
I need to add a specific style on component mount using the addStyle method, but if i mount the component while the mobile portrait device is selected, the style will not be applied on the desktop.

So i need a way that allows me to add style to the desktop without changing the selected device

**Comment by mauriciolcs2** [comment, marked as answer]
I find this workaround, but it's not as sophisticated as I wanted

```
interface AddStyleToDesktopParams {
	component: Component
	style: Record<string, string>
}

function addStyleToDesktop({
	component,
	style,
}: AddStyleToDesktopParams) {
	const devices = component.em.Devices
	const previousSelectedDevice = devices.getSelected()?.id

	devices.select('desktop', { silent: true })

	component.addStyle(style)

	if (previousSelectedDevice) {
		devices.select(previousSelectedDevice as string, {
			silent: true,
		})
	}
}
```

**Reply by artf** [reply]
Check [Css.setRule](https://grapesjs.com/docs/api/css_composer.html#setrule)
